chain/chain/src/store.rs

```diff
@@ -110,9 +110,7 @@ pub trait ChainStoreAccess {
     ) -> Result<ShardChunk, Error> {
         let shard_chunk_result = self.get_chunk(&header.chunk_hash());
         match shard_chunk_result {
-            Err(_) => {
-                return Err(ErrorKind::ChunksMissing(vec![header.clone()]).into());
-            }
+            Err(_) => Err(ErrorKind::ChunksMissing(vec![header.clone()]).into()),
```

Comment:
wait I don't think the transformation here is semantic preserving

Reply:
The result of `match shard_chunk_result {` is returned to `match shard_chunk_result {`
then to ` fn get_chunk_clone_from_header(`

Reply:
The result is preserved, by we can get @matklad opinion on this is you prefer.

Reply:
> The result of match shard_chunk_result { is returned to match shard_chunk_result {
then to fn get_chunk_clone_from_header(

I don't think that is the case. See official rust doc here https://doc.rust-lang.org/rust-by-example/error/result/early_returns.html

Reply:
This is a simpler case than the one in the doc. 
@matklad Which code style should we use from the two above:
A) This is what we have in the code
```
fn multiply(first_number_str: &str, second_number_str: &str) -> Result<i32, ParseIntError> {
   match first_number_str.parse::<i32>() {
        Ok(first_number)  => first_number,
        Err(e) => return Err(e),
    };
}
```
B) This is how we can write it
```
fn multiply(first_number_str: &str, second_number_str: &str) -> Result<i32, ParseIntError> {
   match first_number_str.parse::<i32>() {
        Ok(first_number)  => first_number,
        Err(e) => e,
    };
}
```

@bowenwang1996 No that, there is `let match = `.
If you think that this change is not correct, we should open an issue with `clippy`. https://github.com/rust-lang/rust-clippy

Reply:
@bowenwang1996 Here is the warning from `clippy`:
```
warning: unneeded `return` statement
   --> chain/chain/src/store.rs:114:17
    |
114 |                 return Err(ErrorKind::ChunksMissing(vec![header.clone()]).into());
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove `return`: `Err(ErrorKind::ChunksMissing(vec![header.clone()]).into())`
    |
    = note: `-W clippy::needless-return` implied by `-W clippy::all`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
```

Reply:
I removed this change from the PR, as it doesn't fix the clone() issue.

Reply:
This belongs to a different PR.